### ISSUE TITLE
Typescript support for moduleResolution: node

### ISSUE THREAD

**Issue by redbmk** [auth0/nextjs-auth0#1816]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I'm using `v4.0.0-beta.7` and my tsconfig.json has `"moduleResolution": "node"`. I cannot import types via `@auth0/nextjs-auth0/server` as documented (webpack is able to load it ok). I've tried switching my module resolution to typescript's recommended suggestions ("node16", "nodenext", or "bundler"), however any of those end up breaking imports for other libraries I'm using.

Typescript can import from `@auth0/nextjs-auth0/dist/server`, but then webpack (potentially other bundlers as well) has issues importing from there.

### Describe the ideal solution

Ideally we would be able to import the types even when moduleResolution is set to "node". This worked in v3 and I haven't run into this issue with any other libraries yet.

### Alternatives and current workarounds

A workaround is to create a path alias in tsconfig:

```jsonc
{
  "compilerOptions": {
    // ...
    "paths": {
      "auth0/nextjs-auth0/server": ["node_modules/@auth0/nextjs-auth0/dist/server"],
      // ...
    }
   },
   // ...
}
```

Another possible workaround (untested) is to import from `@auth0/nextjs-auth0/dist/server` and teach the bundler (webpack in this case) how to handle that properly. This seems less ideal having to import from `/dist`.

### Additional context

I was going to file this as a bug, but it requires checking that:

> The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A). *

...which is coded against v3 and doesn't seem to use typescript. It might be worth updating that to be an optional checkmark now that v4 is the main focus

**Comment by guabu** [comment]
Hey @redbmk 👋 Thanks for reporting this! We're now including `typeVersions` which should help TypeScript resolve the types when the `moduleResolution` is set to `node`. This should be available in the next release.

**Comment by fated-x** [comment]
I ran into this same issue after attempting to upgrade my project to use v4.0.0-beta.9. However, the workaround provided by @redbmk didn't work for me. I am on the brink of giving up. Any idea when the next release will be pushed?

**Comment by guabu** [comment]
Hey @fated-x, the next release should be out early next week!

**Comment by nandan-bhat** [comment]
This issue is fixed on `4.0.0-beta.10`. Closing this ticket.
